decorator.rb

```diff
@@ -0,0 +1,11 @@
+require_relative './nameable'
+class Decorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
```

Comment:
- [ ] Please make sure `@nameable` is readable and writable, You can add `attr_accessor :nameable` for that.